packages/bff/src/routers/agreementRouter.ts

```diff
@@ -1,21 +1,80 @@
 import { ZodiosEndpointDefinitions } from "@zodios/core";
 import { ZodiosRouter } from "@zodios/express";
+import { bffApi } from "pagopa-interop-api-clients";
 import {
-  ExpressContext,
   ZodiosContext,
+  ExpressContext,
   zodiosValidationErrorToApiProblem,
 } from "pagopa-interop-commons";
-import { bffApi } from "pagopa-interop-api-clients";
+import { makeApiProblem } from "../model/domain/errors.js";
+import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
+import { fromBffAppContext } from "../utilities/context.js";
+import {
+  emptyErrorMapper,
+  getAgreementByIdErrorMapper,
+  getAgreementsErrorMapper,
+} from "../utilities/errorMappers.js";
+import { agreementServiceBuilder } from "../services/agreementService.js";
 
 const agreementRouter = (
-  ctx: ZodiosContext
+  ctx: ZodiosContext,
+  clients: PagoPAInteropBeClients
 ): ZodiosRouter<ZodiosEndpointDefinitions, ExpressContext> => {
   const agreementRouter = ctx.router(bffApi.agreementsApi.api, {
     validationErrorHandler: zodiosValidationErrorToApiProblem,
   });
+
+  const agreementService = agreementServiceBuilder(clients);
+
   agreementRouter
-    .get("/agreements", async (_req, res) => res.status(501).send())
-    .post("/agreements", async (_req, res) => res.status(501).send())
+    .get("/agreements", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+
+      try {
+        const {
+          consumersIds,
+          eservicesIds,
+          limit,
+          offset,
+          producersIds,
+          showOnlyUpgradeable,
+          states,
+        } = req.query;
+
+        const result = await agreementService.getAgreements(
+          {
+            offset,
+            limit,
+            producersIds,
+            eservicesIds,
+            consumersIds,
+            states,
+            showOnlyUpgradeable,
+          },
+          ctx
+        );
+        return res.status(200).json(result).end();
+      } catch (error) {
+        const errorRes = makeApiProblem(
+          error,
+          getAgreementsErrorMapper,
+          ctx.logger
+        );
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .post("/agreements", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+
+      try {
+        const result = await agreementService.createAgreement(req.body, ctx);
+        return res.status(200).json(result).end();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
```

Comment:
Please add the the error message, is the fourth parameter of this method.
[As in scala](https://github.com/pagopa/interop-be-backend-for-frontend/blob/1.0.x/src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala#L73)